scripts/src/main/resources/scripts/migrations/2023.05.001

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+source "$(dirname "${0}")"/../functions
+
+function doAdaptConfigurationFiles() {
+  configuration_file="${1}"
+  [ -f "${configuration_file}" ] || return
+  cat "${configuration_file}" | awk '
+    /MAVEN_HOME/                  {gsub("MAVEN_HOME","MVN_HOME")}
+    /MVN_HOME/, /maven/           {gsub("maven","mvn")}
+    /MAVEN_VERSION/               {gsub("MAVEN_VERSION","MVN_VERSION")}
+    /SONARQUBE_HOME/              {gsub("SONARQUBE_HOME","SONAR_HOME")}
+    /SONAR_HOME/, /sonarqube/     {gsub("sonarqube","sonar")}
+    /SONARQUBE_VERSION/           {gsub("SONARQUBE_VERSION","SONAR_VERSION")}
+    /INTELLIJ_EDITION_TYPE/       {gsub("INTELLIJ_EDITION_TYPE","INTELLIJ_EDITION")}
+    /INTELLIJ_EDITION/, /c/       {gsub("c","intellij")}
+    /INTELLIJ_EDITION/, /u/       {gsub("u","ultimate")}
+    /ECLIPSE_EDITION_TYPE/        {gsub("ECLIPSE_EDITION_TYPE","ECLIPSE_EDITION")}
+    /ECLIPSE_EDITION/, /java/     {gsub("java","eclipse")}
+    /DOCKERDESKTOP/               {gsub("DOCKERDESKTOP","docker")}
+    {print $0}
+  ' > "${configuration_file}".tmp
+  mv "${configuration_file}".tmp "${configuration_file}"
+  if [ "${configuration_file}" = "${HOME}/devon.properties" ]
+  then
+    doWarning "Due to the changes in the devon.properties file in the user home directory, it is necessary to update all devonfw IDE installations to avoid errors."
+  fi
```

Comment:
actually the warning should only be raised, if something has changed in ~/devon.properties.

Reply:
Does `awk` report via exit code if it did any changes? Otherwise we can diff the tmp file with the original before overwriting it...

Reply:
@hohwille ~/devon.properties is the same as ${HOME}/devon.properties. But I change it to avoid confusions.

Now I start a diff command and move the file in case of changes

Reply:
Not entirely. You need to move this `if` inside the other `if` to ensure that the user does not get the warning for no reason.

Reply:
I don't know why I didn't see that. I have to continue with my studies.